### PR TITLE
issue #393: Usage error should show there is a --help option

### LINKED ISSUE #393: Usage error should show there is a --help option
For instance
```
$ ./scancode 
Usage: scancode [OPTIONS] <input> <output_file>

Error: Missing argument "input".None
```
It would be better if we had something such as:
```
$ ./scancode 
Usage: scancode [OPTIONS] <input> <output_file>
Try 'scancode --help' for help on options and arguments.

Error: Missing argument "input".None
```

### PR DESCRIPTION
**Pull request by rsiemens**
Went ahead and took a crack at this one. If you have suggestions for change please let me know.

### CODE REVIEW

File: click/exceptions.py [hunk 1]
```diff
@@ -51,7 +51,9 @@ def show(self, file=None):
         color = None
         if self.ctx is not None:
             color = self.ctx.color
-            echo(self.ctx.get_usage() + '\n', file=file, color=color)
+            echo(self.ctx.get_usage() + '\nTry "%s %s" for help.\n'
+                 % (self.ctx.command_path, self.ctx.help_option_names[0]),
```
Comment (untitaker) [inline click/exceptions.py]: What if there is no help option (you can disable it with `add_help_option=False`)?

Comment (Nudies) [inline click/exceptions.py]: Good catch, thanks. I'll update the PR this evening.

**Comment by Nudies** [comment]
May have gone a little wild with tests... let me know if there are any other changes. If not I can squash this once approved.

**Comment by untitaker** [comment]
Excellent, thank you!

**Comment by pombredanne** [comment]
@untitaker Thank you Markus!

**Comment by pombredanne** [comment]
@Nudies and thank you ++ of course!

**Pull request merged**